[*ChangeListener] replace AttributesChangedListener with ProviderChan…

zenghuchen-gg opened the pull request:
…geListener


#### Summary

AttributesChangedListener is a duplicate of ProviderChangeListener and there is no need to keep both. Since AttributesChangedListener is for ember and ProviderChangeListener is from DataModel, it makes more sense to keep ProviderChangeListener.

#### Testing

Tested manually by building nordic light examples and the image size is reduced by 100 bytes.
Existing CI is expected to cover that things still work (no extra CI tests are needed/added)

zenghuchen-gg committed:
[*ChangeListener] replace AttributesChangedListener with ProviderChangeListener

AttributesChangedListener is a duplicate of ProviderChangeListener and
there is no need to keep both. Since AttributesChangedListener is for
ember and ProviderChangeListener is from DataModel, it makes more sense
to keep ProviderChangeListener.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=41518) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/project-chip/connectedhomeip?pullRequest=41518) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/project-chip/connectedhomeip?pullRequest=41518) it.</sub>

restyled-commits committed:
Restyled by clang-format

restyled-commits committed:
Restyled by gn

andy31415 commented on src/data-model-providers/codegen/CodegenDataModelProvider_Write.cpp:
```diff
@@ -48,7 +48,7 @@ namespace {
 using Protocols::InteractionModel::Status;
 
-class ContextAttributesChangeListener : public AttributesChangedListener
+class ContextAttributesChangeListener : public DataModel::ProviderChangeListener
```
The original intent of this seems to have been to replace a ProviderChjangeListener to a AttributesChangeListener ... but now they are the same so can we drop this class alltogether?

zenghuchen-gg replied:
Removed in commit 6: https://github.com/project-chip/connectedhomeip/pull/41518/commits/1e7cccae7d914e477bccc74e53f30edaf1e8b8be

zenghuchen-gg committed:
add direct dependency on data-model-provider

zenghuchen-gg committed:
Update comment to reflect the correct usage

zenghuchen-gg committed:
Remove ContextAttributesChangeListener, which is a wrapper of ProviderChangeListener now

zenghuchen-gg committed:
Remove ContextAttributesChangeListener in server-cluster-shim

restyled-commits committed:
Restyled by clang-format

zenghuchen-gg committed:
Replace AttributesChangedListener with ProviderChangedListener in ember functions

restyled-commits committed:
Restyled by clang-format

zenghuchen-gg committed:
include header needed for Providerchangedlistener

andy31415 committed:
Merge branch 'master' into changed_listener_combination

zenghuchen-gg committed:
Apply suggestions from code review

chip::app:: prefix is redundant because the function is defined inside chip::app namespace

Co-authored-by: Boris Zbarsky <bzbarsky@apple.com>

soares-sergio committed:
[no-ccache]

soares-sergio committed:
Merge branch 'master' into changed_listener_combination

soares-sergio committed:
[no-ccache]

andy31415 committed:
Merge branch 'master' into changed_listener_combination
